pkg/restic/common_test.go

```diff
@@ -33,49 +35,84 @@ import (
+	pvbtest2.Spec.Volume = "pvbtest2-abc"
+	pvbtest2.Status.SnapshotID = "123"
+
 	tests := []struct {
-		name        string
-		annotations map[string]string
-		expected    map[string]string
+		name             string
+		podVolumeBackups []*velerov1api.PodVolumeBackup
+		podAnnotations   map[string]string
+		podName          string
+		expected         map[string]string
 	}{
 		{
-			name:        "nil annotations",
-			annotations: nil,
-			expected:    nil,
+			name:           "nil annotations",
+			podAnnotations: nil,
+			expected:       nil,
 		},
 		{
-			name:        "empty annotations",
-			annotations: make(map[string]string),
-			expected:    nil,
+			name:           "empty annotations",
+			podAnnotations: make(map[string]string),
+			expected:       nil,
 		},
 		{
-			name:        "non-empty map, no snapshot annotation",
-			annotations: map[string]string{"foo": "bar"},
-			expected:    nil,
+			name:           "non-empty map, no snapshot annotation",
+			podAnnotations: map[string]string{"foo": "bar"},
+			expected:       nil,
+		},
+		{
+			name:           "has snapshot annotation only, no suffix",
+			podAnnotations: map[string]string{podAnnotationPrefix: "bar"},
+			expected:       map[string]string{"": "bar"},
+		},
+		{
+			name:           "has snapshot annotation only, with suffix",
+			podAnnotations: map[string]string{podAnnotationPrefix + "foo": "bar"},
+			expected:       map[string]string{"foo": "bar"},
 		},
 		{
-			name:        "has snapshot annotation only, no suffix",
-			annotations: map[string]string{podAnnotationPrefix: "bar"},
-			expected:    map[string]string{"": "bar"},
+			name:           "has snapshot annotation, with suffix",
+			podAnnotations: map[string]string{"x": "y", podAnnotationPrefix + "foo": "bar", podAnnotationPrefix + "abc": "123"},
+			expected:       map[string]string{"foo": "bar", "abc": "123"},
 		},
 		{
-			name:        "has snapshot annotation only, with suffix",
-			annotations: map[string]string{podAnnotationPrefix + "foo": "bar"},
-			expected:    map[string]string{"foo": "bar"},
+			name: "has snapshot annotation, with suffix, and also PVB",
+			podVolumeBackups: []*velerov1api.PodVolumeBackup{
+				pvbtest1,
+				pvbtest2,
+			},
+			podName:        testPodName,
+			podAnnotations: map[string]string{"x": "y", podAnnotationPrefix + "foo": "bar", podAnnotationPrefix + "abc": "123"},
+			expected:       map[string]string{"pvbtest1-foo": "bar", "pvbtest2-abc": "123"},
 		},
 		{
-			name:        "has snapshot annotation, with suffix",
-			annotations: map[string]string{"x": "y", podAnnotationPrefix + "foo": "bar", podAnnotationPrefix + "abc": "123"},
-			expected:    map[string]string{"foo": "bar", "abc": "123"},
+			name: "no snapshot annotation, no suffix, but with PVB",
+			podVolumeBackups: []*velerov1api.PodVolumeBackup{
+				pvbtest1,
+				pvbtest2,
+			},
+			podName:  testPodName,
+			expected: map[string]string{"pvbtest1-foo": "bar", "pvbtest2-abc": "123"},
 		},
```

Comment:
let's add some cases for:

- has both PVBs and annotations - PVBs should take precedence
- PVB list includes PVBs for different pods - only the PVBs for the current pod should be returned